### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference when hosting an actor with --enable-api-logging

### ISSUE THREAD

**Issue by darraghjones** [dapr/dapr#4527]
## In what area(s)?

/area runtime

## What version of Dapr?

dapr --version
CLI version: 1.7.0
Runtime version: 1.7.0

## Expected Behavior

API methods to be logged without any errors

## Actual Behavior

When I start my dapr actor host application like this:

 `dapr run --app-id myapp --app-port 5000 --dapr-http-port 3500 --log-level debug --enable-api-logging -- dotnet run`

when an actor is invoked, the following error occurs:

```
time="2022-04-17T19:06:04.5427024+01:00" level=info msg="HTTP API Called: GET /v1.0/metadata" app_id=myapp instance=DESKTOP scope=dapr.runtime.http-info type=log ver=1.7.0

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x50 pc=0x21f45ac]

goroutine 33 [running]:
github.com/dapr/dapr/pkg/grpc.(*server).getGRPCAPILoggingInfo.func1({0x4f75650, 0xc000828ff0}, {0x44c2880, 0xc00010a780}, 0xc0000d4b80, 0xc000d6d440)
        D:/a/dapr/dapr/pkg/grpc/server.go:317 +0xac
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x4f75650, 0xc000828ff0}, {0x44c2880, 0xc00010a780})
        C:/Users/runneradmin/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/dapr/dapr/pkg/diagnostics.(*grpcMetrics).UnaryServerInterceptor.func1({0x4f75650, 0xc000828ff0}, {0x44c2880, 0xc00010a780}, 0xc0000d4b80, 0xc0000d4ba0)
        D:/a/dapr/dapr/pkg/diagnostics/grpc_monitoring.go:181 +0xf8
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x4f75650, 0xc000828ff0}, {0x44c2880, 0xc00010a780})
        C:/Users/runneradmin/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/dapr/dapr/pkg/diagnostics.GRPCTraceUnaryServerInterceptor.func1({0x4f75650, 0xc000828d50}, {0x44c2880, 0xc00010a780}, 0xc0000d4b80, 0xc0000d4bc0)
        D:/a/dapr/dapr/pkg/diagnostics/grpc_tracing.go:73 +0x407
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x4f75650, 0xc000828d50}, {0x44c2880, 0xc00010a780})
        C:/Users/runneradmin/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1({0x4f75650, 0xc000828d50}, {0x44c2880, 0xc00010a780}, 0xc000891bb8, 0x410ab40)
        C:/Users/runneradmin/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34 +0xbf
github.com/dapr/dapr/pkg/proto/internals/v1._ServiceInvocation_CallActor_Handler({0x471ce80, 0xc000d24000}, {0x4f75650, 0xc000828d50}, 0xc000896600, 0xc000797e60)
        D:/a/dapr/dapr/pkg/proto/internals/v1/service_invocation_grpc.pb.go:100 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00058fc00, {0x4fd54f0, 0xc000010a80}, 0xc000152000, 0xc00068e630, 0x770d800, 0x0)
        C:/Users/runneradmin/go/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc00058fc00, {0x4fd54f0, 0xc000010a80}, 0xc000152000, 0x0)
        C:/Users/runneradmin/go/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        C:/Users/runneradmin/go/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        C:/Users/runneradmin/go/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:939 +0x294
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x50 pc=0x21f45ac]

```

## Release Note

RELEASE NOTE:

**Comment by yaron2** [comment]
cc @amulyavarote

**Comment by amulyavarote** [comment]
@darraghjones Hi, I tried to reproduce this error on my end using our sample dotnet actor examples. I did not get this error. Can you please provide more details?
- How are you running this? On your local system? Which OS are you running your application on?

**Comment by yaron2** [comment]
@amulyavarote I think there needs to be a nil for `*info` [here](https://github.com/dapr/dapr/blob/22240790c02d141bfffcf8cd18136bb3a60d280d/pkg/grpc/server.go#L317) and not log if it's empty.

**Comment by darraghjones** [comment]
> @darraghjones Hi, I tried to reproduce this error on my end using our sample dotnet actor examples. I did not get this error. Can you please provide more details?
> 
> * How are you running this? On your local system? Which OS are you running your application on?

I'm running this locally on Windows 10

**Comment by hanxinimm** [comment]
i get the same error

**Comment by yaron2** [comment]
> i get the same error

This is fixed in 1.7.2

**Comment by hanxinimm** [comment]
@yaron2 

what's wrong ?

.\daprd.exe --app-id identity-authentication-webapi-service --components-path ./components --app-protocol http --app-port 20441 --dapr-http-port 20439 --dapr-grpc-port 20438 --metrics-port 20440 --log-level debug --app-max-concurrency -1 
time="2022-05-18T14:31:20.5919444+08:00" level=info msg="starting Dapr Runtime -- version 1.7.2 -- commit 1f8c0abfc73bb14a71435519e56b2aecb4385f01" app_id=identity-authentication-webapi-service instance=WIN-SERVER scope=dapr.runtime type=log ver=1.7.2
time="2022-05-18T14:31:20.5919444+08:00" level=info msg="log level set to: debug" app_id=identity-authentication-webapi-service instance=WIN-SERVER scope=dapr.runtime type=log ver=1.7.2



panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0xa8 pc=0x2558b4a]

[error.zip](https://github.com/dapr/dapr/files/8713865/error.zip)
